### PR TITLE
Adds --flush-interval to set the flush interval of proxy handler

### LINKED ISSUE #135: partially full response buffers never flush
*Expect*

Non-full buffers should flush in a reasonable amount of time

*Observed*

* Partial buffer fills on the response are not flushed to  oidc-proxy client.
* From kubectl client perspective this will often show up as partial json obj... which will fail to decode.
* Pod delete will hang until something terminates the streaming connection (ie LB timeout)
* setting proxyHandler.FlushInterval = -1 resolves the issue

*Symptoms*
*  kubectl delete pod hangs until api server connection is terminated

*Fix*
I set the proxy to flush immediately and the issue resolves. I'm happy to submit a PR if you would like. We could flush immediately, pick a sane value, or make it configurable. 

```golang
// proxy.go: 128
// set up proxy handler using proxy
proxyHandler := httputil.NewSingleHostReverseProxy(url)
proxyHandler.Transport = p
proxyHandler.ErrorHandler = p.Error

// ----------------------------------------------------
// pod delete (anything that calls w/ watch=true ) can hang 
// setting FlushInterval < 0 immediately flushes 
// and solves issue
// ----------------------------------------------------
proxyHandler.FlushInterval = -1
```

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR adds the flag `--flush-interval`. This sets the proxy handler to flush request bodies at the given duration. The default is 50ms. If zero, no flushing is done. If negative, it will flush immediately. The proxy handler will ignore the value is a streaming request is recognized and will flush immediately.

/assign @simonswine 
fixes #135

### CODE REVIEW

**Comment by mkinsley** [comment]
This looks great, thanks a bunch. 50 msec seems like a reasonable default setting.  I'm a bit curious about the comment: 

> The proxy handler will ignore the value is a streaming request is recognized and will flush immediately.

From what I've observed (the root of the issue),  the proxy wasn't ever flushing partial buffers on streaming connections.  The streaming response from ` GET pods watch=true` was where things were hanging.

**Comment by JoshVanL** [comment]
@mkinsley so I'd need to dig into the std library, but seems that in that case, it just wasn't detecting that as a streaming connection. For example, exec is streaming correctly. It looks like it's just not "smart" enough. I could be missing something though, and misunderstanding

**Comment by mkinsley** [comment]
I took a look: [https://golang.org/src/net/http/httputil/reverseproxy.go#L367](url) .   The stdlib is only keying off `Content-Type` to determine if the response is streaming.

If I'm understanding the HTTP RFC (and assuming HTTP/2) , the absence of a `Content-Length` header implies streaming.  In HTTP 1.1 we could look for `transfer-encoding: chunked`.

```
// flushInterval returns the p.FlushInterval value, conditionally
// overriding its value for a specific request/response.
func (p *ReverseProxy) flushInterval(req *http.Request, res *http.Response) time.Duration {
	resCT := res.Header.Get("Content-Type")

	// For Server-Sent Events responses, flush immediately.
	// The MIME type is defined in https://www.w3.org/TR/eventsource/#text-event-stream
	if resCT == "text/event-stream" {
		return -1 // negative means immediately
	}

	// TODO: more specific cases? e.g. res.ContentLength == -1?
	return p.FlushInterval
}
```

File: cmd/app/options/app.go [hunk 1]
```diff
@@ -39,6 +43,12 @@ func (k *KubeOIDCProxyOptions) AddFlags(fs *pflag.FlagSet) *KubeOIDCProxyOptions
 	fs.IntVarP(&k.ReadinessProbePort, "readiness-probe-port", "P", 8080,
 		"Port to expose readiness probe.")
 
+	fs.DurationVar(&k.FlushInterval, "flush-interval", time.Millisecond*50,
+		"Specifies the interval to flush request bodies. If 0ms, "+
+			"no periodic flushing is done. A negative value means to flush "+
+			"immediately after each write. Ignored, and flushed immediately, when "+
```
Comment (munnerz) [inline cmd/app/options/app.go]: Maybe rephrase as "Streaming requests such as 'kubectl exec' will ignore this option."? Took me a few reads over to get it :)
